web/src/main/java/org/springframework/security/web/authentication/preauth/AbstractPreAuthenticatedProcessingFilter.java

```diff
@@ -194,6 +203,11 @@ private void doAuthenticate(HttpServletRequest request, HttpServletResponse resp
 	}
 
 	private boolean requiresAuthentication(HttpServletRequest request) {
+
+		if (!authenticationRequestMatcher.matches(request)) {
```

Comment:
I'm interested in discussing this approach where it augments the existing logic for when authentication is required vs being the entire logic (i.e. the default implementation of RequestMatcher copies the logic from this method). What are your thoughts on each approach?

My concern about augmenting the existing logic is that there is no way to change the existing logic.

Reply:
You mean performing request matcher check in `requiresAuthentication()` method vs performing the check in `doFilter()`, so that entire logic of this filter may be skipped if request doesn't match?

```java
public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
    throws IOException, ServletException {
    
    if(!requiresAuthenticationRequestMatcher.matches(request)){
      chain.doFilter(request, response);
      return;
    }
    
    // existing logic from here
    
    if (!requiresAuthentication(request, response)) {
      chain.doFilter(request, response);
      return;
    }
    ...
}
```


Currently, `requiresAuthentication()` method on this class `AbstractPreAuthenticatedProcessingFilter` (APAPF) is `private`, so where to put the request-matcher check doesn't really affect any other existing sub classes. So, behavior-wise it is same.

The similar class `AbstractAuthenticationProcessingFilter`(AAPF), the request-matcher check is in `requiresAuthentication()` which is a `protected` method.

So, to align the implementation with AAPF, then request-matcher check is in `requiresAuthentication()`.
If we consider request-matcher check is the first-check even before `requiresAuthentication()`, then the check can go into `doFilter()`(shown in above snippet), and maybe update the AAPF as well (updating AAPF will be a behavioral change).

My preference is put the request-matcher check in `requiresAuthentication()` as in current PR which aligns with AAPF, then also make `requiresAuthentication()` protected, so that sub-class can modify the behavior, even it is inheritance. (Yes, I know spring is for composition, but since this is an abstract class which is meant to be inherited and may be overridden.)


> My concern about augmenting the existing logic is that there is no way to change the existing logic.

Currently the method is `private`, adding all-match request-matcher should have no impact to existing logic.

Reply:
Yes. Since the default implementation is internal it would still be able to refer to the member variables.

Reply:
OK, updated the PR.
Moved the existing auth check logic into the new request matcher.